### ISSUE TITLE
ModuleNotFoundError: No module named 'socketserver'

### ISSUE THREAD

**Issue by minchinweb** [anxdpanic/plugin.video.youtube#114]
### Context

Please provide any relevant information about your setup

* Add-on Version: 6.8.10+matrix.1
* Kodi Version: 19
* Kodi GUI Language: English
* Operating System: Windows 10
* Operating System Language: Canadian English

------

### Expected Behavior

No errors. Plugin should work.

------

### Current Behavior

Throws an error every time I start Kodi or try to open the plugin. I believe it is also throwing the same error when I try to watch a trailer on Youtube.

```
2021-04-12 10:07:02.847 T:4208    ERROR <general>: XFILE::CDirectory::GetDirectory - Error getting plugin://plugin.video.youtube/
2021-04-12 10:07:02.880 T:10700   ERROR <general>: CGUIMediaWindow::GetDirectory(plugin://plugin.video.youtube/) failed
2021-04-12 10:08:21.163 T:4584     INFO <general>: CActiveAESink::OpenSink - initialize sink
2021-04-12 10:08:34.515 T:11164    INFO <general>: CServiceAddonManager: failed to stop plugin.video.youtube (may have ended)
2021-04-12 10:08:34.761 T:10700 WARNING <general>: CGUIMediaWindow::OnMessage - updating in progress
2021-04-12 10:08:49.156 T:10868    INFO <general>: CAddonMgr::ADDON::CAddonMgr::FindAddon: plugin.video.youtube v6.8.10+matrix.1 installed
2021-04-12 10:08:49.161 T:1852     INFO <general>: initializing python engine.
2021-04-12 10:08:49.302 T:10700 WARNING <general>: CGUIMediaWindow::OnMessage - updating in progress
2021-04-12 10:08:49.818 T:1852    ERROR <general>: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                                    - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                                   Error Type: <class 'ModuleNotFoundError'>
                                                   Error Contents: No module named 'socketserver'
                                                   Traceback (most recent call last):
                                                     File "C:\Users\William\AppData\Roaming\Kodi\addons\plugin.video.youtube\resources\lib\startup.py", line 11, in <module>
                                                       from youtube_plugin.kodion import service
                                                     File "C:\Users\William\AppData\Roaming\Kodi\addons\plugin.video.youtube\resources\lib\youtube_plugin\kodion\__init__.py", line 18, in <module>
                                                       from .abstract_provider import AbstractProvider
                                                     File "C:\Users\William\AppData\Roaming\Kodi\addons\plugin.video.youtube\resources\lib\youtube_plugin\kodion\abstract_provider.py", line 20, in <module>
                                                       from . import utils
                                                     File "C:\Users\William\AppData\Roaming\Kodi\addons\plugin.video.youtube\resources\lib\youtube_plugin\kodion\utils\__init__.py", line 19, in <module>
                                                       from .http_server import get_http_server, is_httpd_live, get_client_ip_address
                                                     File "C:\Users\William\AppData\Roaming\Kodi\addons\plugin.video.youtube\resources\lib\youtube_plugin\kodion\utils\http_server.py", line 10, in <module>
                                                       from six.moves import BaseHTTPServer
                                                     File "C:\Users\William\AppData\Roaming\Kodi\addons\script.module.six\lib\six.py", line 92, in __get__
                                                       result = self._resolve()
                                                     File "C:\Users\William\AppData\Roaming\Kodi\addons\script.module.six\lib\six.py", line 115, in _resolve
                                                       return _import_module(self.mod)
                                                     File "C:\Users\William\AppData\Roaming\Kodi\addons\script.module.six\lib\six.py", line 82, in _import_module
                                                       __import__(name)
                                                     File "C:\Program Files\Kodi\system\python\Lib\http\server.py", line 102, in <module>
                                                       import socketserver
                                                   ModuleNotFoundError: No module named 'socketserver'
                                                   -->End of Python script error report<--

```

------

### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Make sure the addon is installed.
2. Go to the YouTube addon page in the Kodi settings.
3. Select the "Open" button at the bottom.

------

### Log

Please include a complete [debug log](https://kodi.wiki/view/Log_file).

See above.

------

### Additional Information

I'm surprised that `six` would be the source of the error... maybe an upstream bug report needs to be opened?

Or is Kodi running a limited version of the Python standard library that doesn't include `socketserver`?

------

**Comment by titobrasolin** [comment]
> * Kodi Version: 19
> * Operating System: Windows 10

Did you upgrade from a previous version of Kodi? If so, the "SocketServer.py" file was probably overwritten but not renamed to "socketserver.py" (Windows is not case sensitive).

This is exactly what happened to me, so I uninstalled and reinstalled Kodi and now the YouTube plugin works perfectly.

**Comment by minchinweb** [comment]
Thank you for the explanation and help! That's a ... rather ... unsatisfying answer. So I raised a ticket upstream with Kodi. I'll rename the files in questions (`queue.py` is similar) or reinstall as needed.

You can probably close this ticket, as the issue is upstream.

**Comment by minchinweb** [comment]
It seems this is only an issue if you do a silent upgrade (which chocolately does). The silent upgrade option doesn't actually delete the old installation. Kodi has accepted the tickets, so maybe this will get fixed for 19.1...
